### ISSUE TITLE
Faster Calculations via in-place example

### ISSUE THREAD

**Issue by finmod** [JuliaDiffEq/DifferentialEquations.jl#46]
While running this example, the following error comes up:

LoadError: BoundsError
while loading In[3], in expression starting on line 2

 in getindex at .\number.jl:21 [inlined]
 in f₁(::Rational{Int64}, ::Array{Float64,1}) at .\In[1]:3
 in f(::Rational{Int64}, ::Array{Float64,1}) at .\In[1]:6
 in (::DifferentialEquations.##560#568{DifferentialEquations.ODEProblem{Array{Float64,1},Float64}})(::Rational{Int64}, ::Array{Float64,1}, ::Array{Float64,1}) at C:\Users\Denis\.julia\v0.5\DifferentialEquations\src\ode\ode_solve.jl:79
 in ode_solve(::DifferentialEquations.ODEIntegrator{:RK4,Array{Float64,1},Float64,2,Rational{Int64},Float64,Array{Float64,1},Array{Float64,1}}) at C:\Users\Denis\.julia\v0.5\DifferentialEquations\src\ode\ode_integrators.jl:110
 in #solve#559(::Array{Any,1}, ::Function, ::DifferentialEquations.ODEProblem{Array{Float64,1},Float64}, ::Array{Int64,1}) at C:\Users\Denis\.julia\v0.5\DifferentialEquations\src\ode\ode_solve.jl:190
 in (::DifferentialEquations.#kw##solve)(::Array{Any,1}, ::DifferentialEquations.#solve, ::DifferentialEquations.ODEProblem{Array{Float64,1},Float64}, ::Array{Int64,1}) at .\<missing>:0

This BoundsError message appears in other examples: Solving the Giere-Meinhardt...

Thanks

**Comment by ChrisRackauckas** [comment]
Are you on master? Those should all be fixed by changing the Delta t to floating point instead of Rational, but it should work with Rational. I'll look into this right now.

**Comment by finmod** [comment]
Yes, I am on master. Have not considered any other version since DifferentialEquations appeared

**Comment by ChrisRackauckas** [comment]
Oh...

I need to update the examples. Everything is defined with `(t,u,du)` instead of `(du,u,t)` now. I did the `(du,u,t)` way because it matched Julia's notation of the in-place array always being first. But I changed to `(t,u,du)` since that's the Fortran convention and so it helps with FFI, and it matches ODE.jl so it makes it easier for compatibility. If you change the function definitions to use `(t,u)` and (t,u,du)` then it should all work. If you use the function definition macro, it should work without reference to this ordering.

Thanks for notifying me. I'll update the examples.
